### PR TITLE
[5.5] Supporting Laravel-Mix PR #1246, HMR w/ different Hosts and Ports

### PR DESCRIPTION
**Pull request by sifex**
Part One: #21545

This PR changes the 'mix' helper function to take URLs coming from the `hot` file. This feature should be fully backwards compatible with older Laravel Mix & Laravel installs.

Supporting Laravel-Mix Pull Request JeffreyWay/laravel-mix#1246, allowing Hot Module Reloading with custom hosts and ports, allowing hot-module reloading across networks.

~~Please don't merge straight away, I'm awaiting a reply to JeffreyWay/laravel-mix#1246~~

JefferyWay merged JeffreyWay/laravel-mix#1246. Should be good to go 

Let me know if you require tests for this code change,

Edit: Just realised that there's no other use of `Str::` in the class. Let me know if you want me to use the native php string functions.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Will let @JeffreyWay chime in on this.

**Comment by taylorotwell** [comment]
Can you explain with this does in kinda simple terms? Sorry, I'm not *super* up to speed on the latest front-end stuff. :smile:

File: src/Illuminate/Foundation/helpers.php [hunk 1]
```diff
@@ -555,6 +555,12 @@ function mix($path, $manifestDirectory = '')
         }
 
         if (file_exists(public_path($manifestDirectory.'/hot'))) {
+            $url = file_get_contents(public_path($manifestDirectory.'/hot'));
+
+            if (Str::startsWith($url, 'http://') || Str::startsWith($url, 'https://')) {
```
Comment (ntzm) [inline src/Illuminate/Foundation/helpers.php]: This line can be shortened to:

```php
if (Str::startsWith($url, ['http://', 'https://'])) {
```

**Comment by JeffreyWay** [comment]
Yep, this is fully backward compatible and fine to merge.

File: src/Illuminate/Foundation/helpers.php (hunk 1 above)
Comment (sifex) [reply src/Illuminate/Foundation/helpers.php]: Nice find

**Pull request merged**